code/game/machinery/asteroid_cannon/ads_extension.dm

```diff
@@ -117,4 +117,11 @@
 	if(!istype(holder, /obj/structure/asteroidcannon))
 		return FALSE
 	gun = holder
-	START_PROCESSING(SSfastprocess, src)
+	wake_up()
+
+
+
+
+/datum/extension/asteroidcannon/proc/wake_up()
+	if (!is_processing)
```

Comment:
as you demonstrated, this check is unnecessary since it already does that check as part of START_PROCESSING and STOP_PROCESSING

so you can remove that and others like it